src/common/components/elements/ProfilePreviewCard.tsx

```diff
@@ -0,0 +1,199 @@
+import { useAnalytics } from '@/common/context/AnalyticsProvider';
+import { getFallbackImage } from '@/modules/utils/image';
+import { showFirstAndLastFour } from '@/modules/utils/string';
+import { ProfilePreviewData } from '@/types/types';
+import { useAnchorWallet, useConnection } from '@solana/wallet-adapter-react';
+import classNames from 'classnames';
+import Link from 'next/link';
+import { useState, useEffect, useCallback, FC, VFC } from 'react';
+import { useProfilePreviewLazyQuery, useIsXFollowingYLazyQuery } from 'src/graphql/indexerTypes';
+import { AvatarImage } from './Avatar';
+import { FollowUnfollowButton } from './FollowUnfollowButton';
+
+export interface ProfilePreviewProps {
+  address: string;
+  data?: ProfilePreviewData;
+  onInsufficientData: (address: string) => void;
+}
+
+export default function ProfilePreview(props: ProfilePreviewProps): JSX.Element {
+  const { track } = useAnalytics();
+  const [dataQuery, dataQueryContext] = useProfilePreviewLazyQuery();
```

Comment:
Here is a spot where data fetching should be pulled into the view can look into apollo fragments if you want it to define what data its needs from the api.

https://www.apollographql.com/docs/react/data/fragments/